### ISSUE TITLE
Vendor.js bundle is 1.5 mb already. How to reduce it ?

### ISSUE THREAD

**Issue by ibudisteanu** [vuejs/vue-hackernews-2.0#372]
I am using the Vue hackernews as a boilerplate for a project. I just realized the vendor.js bundle size is over 1.5 mb in total it contains many packages that the browser doesn't need like "mime-db", "bn.js" (big number), "bluebird.js", "tweetnacl", "elliptic", etc. Some of them are used by the express which is used only on the server side, but most of the packages are included to the browser vendor.js bundle.

I installed the Webpack bundle analyzer and I saw there are so many redundancies as well. Any help to remove them out from the vue-hackernews 2.0 boilerplate. 

![image](https://user-images.githubusercontent.com/17750489/65395199-5c9fe780-dda0-11e9-84c3-446b989dd3e9.png)

**Comment by ibudisteanu** [comment]
I was able to reduce the entire chunks including vendor.js to 1.5 mb

But still, the vendor.js file has 1.4 mb and I have no clue why? Any explanation with mime-db and mime-type is required in frontend ?

![image](https://user-images.githubusercontent.com/17750489/65438174-b7cbeb80-de2d-11e9-94f0-c78fd88bdd76.png)

**Comment by sirlancelot** [comment]
I suspect several of those are deep dependencies of the firebase module, probably?

**Comment by ibudisteanu** [comment]
Hi Thanks! I checked for Firebase. My code doesn't use firebase. I searched throughout all the files. Firebase was also removed from the package.json dependencies.

I can only see that firebase is indeed installed and shown in the yarn.lock but otherwise, my code doesn't use firebase anymore.

![image](https://user-images.githubusercontent.com/17750489/65463207-127e3b00-de60-11e9-856a-1c4595da9ef6.png)

**Comment by sirlancelot** [comment]
Also keep in mind Webpack's tree-shaking only applies to production bundles. It's not perfect, but it should cut down on the bloat significantly.

This repository is also incredibly outdated, there's new versions of almost every dependency, including Webpack. I suggest taking this discussion to a more appropriate venue, such as Discord or Forum.

**Comment by ibudisteanu** [comment]
Today I decided to take another look at this issue. It seems it was my fault by installing request-promise npm. That library alone is over 1 MB. After I replaced request-promise with node-fetch, I got less than 380kb in total.

![image](https://user-images.githubusercontent.com/17750489/66319122-3bed9b00-e925-11e9-849f-c9df4cd71c60.png)

![image](https://user-images.githubusercontent.com/17750489/66319157-490a8a00-e925-11e9-8d2b-8d521a1f49d6.png)
